sonar-dotnet-shared-library/src/main/java/org/sonar/plugins/dotnet/tests/Coverage.java

```diff
@@ -57,6 +61,22 @@ void addHits(String file, int line, int hits) {
     oldHitsByLine[i] += hits;
   }
 
+  public void addBranchCoverage(String file, BranchCoverage branchCoverage){
+    List<BranchCoverage> branchCoverages = branchCoverageByFile.computeIfAbsent(file, k -> new ArrayList<>());
+
+    // If there are multiple branch coverage entries per line these need to be merged; otherwise SQ/SC will display only
+    // one of them.
+    Optional<BranchCoverage> existingCoverage = branchCoverages.stream()
+      .filter(coverage -> coverage.getLine() == branchCoverage.getLine())
+      .findAny();
```

Comment:
this `findAny` means you get one result ... but from what I understand, you can have more than one already, right?

or you should not because here you are always merging?

Reply:
There can be only one due to merging.